### ISSUE TITLE
Caches disappear on map switch with new mapsforge beta

### ISSUE THREAD

**Issue by Matze2** [cgeo/cgeo#6933]
##### Detailed steps causing the problem:
- Open live map with an offline OSM map (Freizeitkarte or Openandromaps with freizartkarte-v5.xml theme)
- Switch to Google Maps; caches are still there
- Switch back to OSM map

##### Actual behavior after performing these steps:
Caches disappeared. Whether they are saved or not is irrelevant.
The OSM theme used seems to be irrelevant, too.

##### Other strange behavior (not sure if this is all the same problem or we need separate issues)
- Zooming out to "2km" brings the caches back, zooming in to "1km" make them disappear again.
- If automatic rotation is active and the device is rotated then the map zooms out by 3 levels. Rotating back rotates again by 3 levels. At the c:geo crashes.
- Zoom out button is disabled at "10km", but using two fingers it is possible to zoom out to "5000km" or even to "10000km"

All tested using 2018-02-14-NB1.

**Comment by Lineflyer** [comment]
I thought this appears when changing between two offline maps.

The case of switching between Google and mapsforge now simply starts the other activity....but this should also populate the map...mmh.

**Comment by Matze2** [comment]
Just checked. Yes it is just switching. It also happens when switching between two offline maps.

**Comment by pstorch** [comment]
The map crashes could be related to: https://github.com/mapsforge/mapsforge/issues/957
At least this is what I see in the logs when switching the map source.

**Comment by Lineflyer** [comment]
I used the new mapsforge very intensivly to prepare a cache trip last weekend.
I also recognized a problem with disappearing caches (without any switching between map providers).

It is somehow like parts of the map are being populated with caches and a small move of the map makes them disappear in a way which looks randomly. Moving the map again makes them reappear and then again other caches disappear once the loading is finished.

@Matze2 Is this, what you also described above?
